app/models/storage.rb

```diff
@@ -77,6 +78,10 @@ def ext_management_systems
       :host_storages => {:storage_id => id}).distinct.to_a
   end
 
+  def storage_cluster
```

Comment:
I feel like this method should return a StorageCluster instance.  Then you can `delegate :name, :to => :storage_cluster, :prefix => true`